dist-persist/wbstack/src/Internal/ApiWbStackSiteStatsUpdate.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WBStack\Internal;
+
+/**
+ * This should update the site_stats table
+ */
+class ApiWbStackSiteStatsUpdate extends \ApiBase {
+    public function mustBePosted() {return true;}
+    public function isWriteMode() {return true;}
+    public function isInternal() {return true;}
+    public function execute() {
+        global $IP;
+
+        @set_time_limit( 60*5 ); // 5 mins maybe D:
+		@ini_set( 'memory_limit', '-1' ); // also try to disable the memory limit? Is this even a good idea?
```

Comment:
I'm not really a fan of using these functions here, and also suppressing errors - I guess it's there so the output doesn't get garbled and the request doesn't time out, so it makes sense but I think it is a bit of bad practice.

That said, I don't have an alternative approach at hand right now so there's that :sweat_smile:

Reply:
So this is used(copy-pasta) for most of these endpoints that do any heavy lifting. If you want we could try removing the memory limit for example, not sure if it's needed. The time limit i think is required though.

What would be suppressing errors?

Reply:
> What would be suppressing errors?

The `@` operator in front of function calls

> So this is used(copy-pasta) for most of these endpoints that do any heavy lifting. 

Ah I wasn't aware, then it's more of a refactoring thing I guess, I'm okay with that :+1:

Reply:
Huh, I did not know that was for error control. TIL!